include/internal/ktls.h

```diff
@@ -96,11 +114,79 @@ static ossl_inline int ktls_send_ctrl_message(int fd, unsigned char record_type,
     return sendmsg(fd, &msg, 0);
 }
 
+#   ifdef OPENSSL_NO_KTLS_RX
+
 static ossl_inline int ktls_read_record(int fd, void *data, size_t length)
 {
     return -1;
 }
 
+#   else /* !defined(OPENSSL_NO_KTLS_RX) */
+
+/*
+ * Receive a TLS record using the tls_en provided in ktls_start.  The
+ * kernel strips any explicit IV and authentication tag, but provides
+ * the TLS record header via a control message.  If there is an error
+ * with the TLS record such as an invalid header, invalid padding, or
+ * authentication failure recvmsg() will fail with an error.
+ */
+static ossl_inline int ktls_read_record(int fd, void *data, size_t length)
```

Comment:
Over time the "inline" functions in this file seem to have been getting longer. I wonder if a refactor is in order to no longer declare these longer functions as "inline".

Reply:
As in a ssl/ktls.c only included if the ktls option is defined?  We could move the longer functions there (ktls_check_supported_cipher and ktls_configure_crypto).  That would sidestep the somewhat ugly hack of checking for OSSL_SSL_LOCAL_H in ktls.h.  I think the other inline functions are still relatively short and are used in libcrypto so might be best left in ktls.h.

Reply:
> As in a ssl/ktls.c only included if the ktls option is defined?

Yes - something like that is what I had in mind.

>  think the other inline functions are still relatively short and are used in libcrypto so might be best left in ktls.h.

That's fine. Its mainly the longer ones which just look wrong to me as inline functions.

Reply:
Ok, I've done this.